### PR TITLE
connection server to frontend

### PR DESCRIPTION
**Pull request by marcushohlbein**

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
-import data from '../data/products.json'
+//import json from '../data/products.json'
```
Comment (macfrei) [inline client/src/components/App.js]: You don't need it anymore :)

File: client/src/components/App.js [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+            {data?.map(product => (
+              <ProductListing key={product._id} product={product} />
             ))}
           </ProductsContainer>
         </Route>
-        <Route
-          path="/:styleId"
-          render={props => <ProductDetail {...props} />}
-        />
+        <Route path="/:styleId" children={<ProductDetail />} />
```
Comment (macfrei) [inline client/src/components/App.js:31]: Why not `productId` or just `id`? It is the id of the product, right? `styleId` sounds a bit like it has something to do with styling/css. But that's just my personal opinion.

File: client/src/components/ProductDetail/ProductDetail.js [hunk 3]
```diff
@@ -32,15 +50,24 @@ export default function ProductDetail({ location }) {
...
   media: PropTypes.array,
   shops: PropTypes.array,
 }
 
 const ProductContainer = styled.main`
-  display: grid;
-  gap: 20px;
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
```
Comment (macfrei) [inline client/src/components/ProductDetail/ProductDetail.js]: Be careful with `gap` for flexbox as it is not supported in all browsers yet. Also, you could write it with grid and save yourself one line of code 😃 
```suggestion
  display: grid;
  gap: 10px;
```

File: routes/error.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+module.exports = (err, req, res, next) => {
+  console.log(err.message)
+  res.json({ error: err.message })
```
Comment (macfrei) [inline routes/error.js:3]: I think you'll need some error handling in your frontend as well.

**Review by macfrei — COMMENTED** [review]
Well done, added a few comments :)

File: client/src/components/App.js (hunk 2 above)
Comment (marcushohlbein) [reply client/src/components/App.js:31]: styleId is a fancy word for SKU. it's unique but ist not like the id

**Review by mhstrkmp — COMMENTED** [review]
lgtm 🚀

File: client/src/components/App.js (hunk 2 above)
Comment (macfrei) [reply client/src/components/App.js:31]: Learned something new! Thanks :)

**Pull request merged**